windows/RNDocumentPicker/RCTDocumentPickerModule.cs

```diff
@@ -0,0 +1,188 @@
+	private static readonly String FIELD_TYPE = "type";
+	private static readonly String FIELD_SIZE = "size";
+
+        public RCTDocumentPickerModule(ReactContext reactContext)
+            : base(reactContext)
+        {
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "RNDocumentPicker";
+            }
+        }
+
+        public override void Initialize()
+        {
+            Context.AddLifecycleEventListener(this);
+        }
+
+        public void OnSuspend()
+        {
+            _isInForeground = false;
+        }
+
+        public void OnResume()
+        {
+            _isInForeground = true;
+        }
+
+        public void OnDestroy()
+        {
+        }
+
+        [ReactMethod]
+        public void pick(JObject options, IPromise promise)
+        {
+            try
+            {
+                FileOpenPicker openPicker = new FileOpenPicker();
+                openPicker.ViewMode = PickerViewMode.Thumbnail;
+                openPicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+                // Get file type array options
+                var fileTypeArray = options.Value<JArray>(OPTION_TYPE);
+                // Init file type filter
+                if (fileTypeArray != null && fileTypeArray.Count > 0)
+                {
+                    foreach (String typeString in fileTypeArray)
+                    {
+                        List<String> types = typeString.Split(' ').ToList();
+                        foreach (String type in types)
+                        {
+                            if (Regex.Match(type, "(^[.]+[A-Za-z0-9]*$)|(^[*]$)").Success)
+                            {
+                                openPicker.FileTypeFilter.Add(type);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    openPicker.FileTypeFilter.Add("*");
+                }
+
+                RunOnDispatcher(async () =>
+                {
+                    try
+                    {
+                        if (_isInForeground)
+                        {
+                            var isMultiple = options.Value<bool>(OPTION_MULIPLE);
+                            if (isMultiple)
+                            {
+                                await PickMultipleFileAsync(openPicker, promise);
+                            }
+                            else
+                            {
+                                await PickSingleFileAsync(openPicker, promise);
+                            }
+                        }
+                        else
+                        {
+                            _pendingPicker = openPicker;
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        promise.Reject(E_FAILED_TO_SHOW_PICKER, E_FAILED_TO_SHOW_PICKER);
```

Comment:
Get the error message from the actual exception instead of hiding it from the developer.